backend/clubs/views.py

```diff
@@ -4786,19 +4809,20 @@ def cart(self, request, *args, **kwargs):
         sold_out_count = 0
 
         replacement_tickets = []
-        for gone_ticket in tickets_to_replace:
+        for ticket_class in tickets_to_replace.values("type", "event").annotate(
+            count=Count("id")
+        ):
             # We don't need to lock since we aren't updating holder/owner
-            ticket = Ticket.objects.filter(
-                event=gone_ticket.event,
-                type=gone_ticket.type,
+            tickets = Ticket.objects.filter(
+                event=ticket_class["event"],
+                type=ticket_class["type"],
                 owner__isnull=True,
                 holder__isnull=True,
-            ).first()
+            ).exclude(id__in=cart.tickets.all())[: ticket_class["count"]]
```

Comment:
nit: I think we can fetch this once, outside the loop. something like `cart.tickets.values_list('id', flat=True)`